straxen/plugins/peaks/_peak_s1_position_base.py

```diff
@@ -0,0 +1,94 @@
+Author and maintainer: Matteo Guida (guidam@mpi-hd.mpg.de)"""
+
+import strax
+import straxen
+import numpy as np
+from warnings import warn
+import time
+export, __all__ = strax.exporter()
+
+@export
+class PeakS1PositionBase(strax.Plugin):
+    """
+    Base pluging for S1 position reconstruction at peak level.
+    Provides x_algorithm, y_algorithm, z_algorithm for all peaks > than
+    min-reconstruction area based on both the top and bottom array.
+    """
+    __version__ = '0.0.0'
+    depends_on = ('peaks',)
+
+    algorithm = None
+    compressor = 'zstd'
+    parallel = True  # can set to "process" after #82
+
+    min_s1_area_s1_posrec = straxen.URLConfig(
+        help='Skip reconstruction if area (PE) is less than this',
+        default=1000, infer_type=False, )
+
+    def infer_dtype(self):
+        if self.algorithm is None:
+            raise NotImplementedError(f'Base class should not be used without '
+                                      f'algorithm as done in {__class__.__name__}')
+
+    def infer_dtype(self):
+        if self.algorithm is None:
+            raise NotImplementedError(f'Base class should not be used without '
+                                      f'algorithm as done in {__class__.__name__}')
+        dtype = [('x_' + self.algorithm, np.float32,
+                  f'Reconstructed {self.algorithm} S1 X position (cm), uncorrected'),
+                 ('y_' + self.algorithm, np.float32,
+                  f'Reconstructed {self.algorithm} S1 Y position (cm), uncorrected'),
+                 ('z_' + self.algorithm, np.float32,
+                  f'Reconstructed {self.algorithm} S1 Z position (cm), uncorrected')
+                  ]
+
+        dtype += strax.time_fields
+        return dtype
+    
+    def get_tf_model(self):
+        """
+        Simple wrapper to have several tf_peak_model_s1_cnn, ..
+        point to this same function in the compute method
+        """
+        model = getattr(self, f'tf_peak_model_{self.algorithm}', None)
+        if model is None:
+            warn(f'Setting model to None for {self.__class__.__name__} will '
+                 f'set only nans as output for {self.algorithm}')
+        if isinstance(model, str):
+            raise ValueError(f'open files from tf:// protocol! Got {model} '
+                             f'instead, see tests/test_s1_posrec.py for examples.')
+        return model
+
+    def compute(self, peaks):
+        result = np.ones(len(peaks), dtype=self.dtype)
+        result['time'], result['endtime'] = peaks['time'], strax.endtime(peaks)
+
+        result['x_' + self.algorithm] *= float('nan')
+        result['y_' + self.algorithm] *= float('nan')
+        result['z_' + self.algorithm] *= float('nan')
+
+        model = self.get_tf_model()
+
+        # Reconstruct position only for large peaks, otherwise severe inaccuracy.
+        peak_mask = (peaks['area_per_channel'].sum(axis=1) > self.config['min_s1_area_s1_posrec'])&(peaks['type']<2)
```

Comment:
This is a good point, I would say there is not a well-defined motivation. In the S2 version `_peak_positions_base.py` there is no filter: the predictions are made on peaks classified as 0: unknown, 1: S1, 2: S2. 

Not putting any filters, in this case, seemed unreasonable to me. I thought: since there is already the area>1000 filter, we also make predictions on 0 in case there has been some classification difficulty which perhaps can be detected later. But we could also directly put `peaks['type']==1` if you think is better.

Reply:
Let's just stay with the current code.